### DISCUSSION TITLE
When to expect progress reports in results.txt

### DISCUSSION THREAD

**Discussion by fuglede** [preda/gpuowl#301] [Q&A] [answered]
From a cursory look at the source code, it looks like there's support for writing PRP progress reports to `results.txt`. I'm using AutoPrimeNet.py to get assignments, and am running `./prpll -dir /folder/with/assignments` to do the work. After many hours of work and about 1% completion, `results.txt` is still empty. Is this expected, and am I perhaps simply impatient? Thanks!

**Comment by tdulcet** [comment]
What do you mean by "PRP progress reports to `results.txt`"? AutoPrimeNet submits assignment progress to the PrimeNet server, which would be visible on several mersenne.org pages. However, PRPLL is not yet compatible with the PrimeNet server, so you would need to use GpuOwl for this to work.

https://github.com/preda/gpuowl/blob/73b96ebacf2a23eb052b7d13848ffe7f28c9cc3b/src/Args.cpp#L102

**Reply by fuglede** [reply]
> However, PRPLL is not yet compatible with the PrimeNet server, so you would need to use GpuOwl for this to work.

Ah, thanks, I see. Because this repository is called `gpuowl`, and the `README.md` is about PRPLL, I might have expected the two to be synonymous. What I have been doing is use AutoPrimeNet to generate a `worktodo.txt`, then use `prpll -dir` on the same directory to do work, but maybe that's not the intended approach.

**Reply by tdulcet** [reply, marked as answer]
PRPLL is a new in development version of GpuOwl. What you described may be the intended approach if/when PRPLL is finished and PrimeNet server compatible. Until then, I would recommend using GpuOwl, the latest version of which can be found on the [`gpuowl` branch](https://github.com/preda/gpuowl/tree/gpuowl) of this repository.

**Reply by fuglede** [reply]
Gotcha, thanks!

**Reply by fuglede** [reply]
Works like a charm!

**Comment by preda** [comment]
I would recommend to use PRPLL instead of GpuOwl 7.x or 6.x, as I consider PRPLL an improvement in every way except not having P-1. If P-1 is a requirement for the user, P-1 can be found in older GpuOwl.

Also, I'd like to point out that I'm fine with users making their own choice and decision about what to use. Also @tdulcet is free to express his personal oppinions, which are just that.

**Reply by tdulcet** [reply]
Yes, PRPLL has several exciting new features, including auto tuning and the return of support for LL DC tests. I know many users including myself are looking forward to being able to use it for our production work. However, as you know, PRPLL also removed PrimeNet server compatibility, so users that want to use a modern and full featured PrimeNet client with support for assignment progress reporting for example must continue to use GpuOwl. This is why the PrimeNet administrators recommend that users continue to use GpuOwl and is presumably one of the reasons why your `--help` output does not recommend using PRPLL for production work:
https://github.com/preda/gpuowl/blob/9c8a2b8a1712fd1aca73bb33a4427a68490c5978/src/Args.cpp#L102
Unfortunately, this is not my "personal opinion", but fact. I am still very hopeful that we can all find a solution to restore PrimeNet server compatibility to PRPLL. Please do let me know if there is anything I can do to help with this.

**Comment by preda** [comment]
@fuglede to answer your initial question,
results.txt is only updated once a test (let's say PRP) is completed, done. That's what a result is, the outcome of the PRP test.

**Comment by fuglede** [comment]
As far as I understand, the various PrimeNet tools also support using `reports.txt` as a placeholder for progress, which might look e.g. as
```
[Fri Oct 25 17:16:57 2024]
Trial factoring M330767207 to 2^76 is 1.70% complete.
```
I see that GpuOwl, as of the `gpuowl` branch, does do something else to tell AutoPrimeNet that progress is being made, though. Whatever the mechanism is, it seems like the current `main` doesn't do the same thing, because I got no progress reports from that one. Progress reports, I suppose, are mainly useful for very long tasks, e.g. PRP tests on consumer hardware, where you want to keep an assignment alive.

**Reply by fuglede** [reply]
Okay, `autoprimenet.py` just works by parsing `gpuowl.log`, so assuming the format hasn't changed, that seems like it would be straightforward to extend to PRPLL.

**Reply by tdulcet** [reply]
Some older GIMPS software do output messages to the results file that may include the progress, but those are purely for informational purposes and GpuOwl or PRPLL also do not support this. Only the JSON formatted results in the results file are sent to the server.

Yes, for GpuOwl, the AutoPrimeNet client determines the progress primarily from parsing the `gpuowl.log` file. For PRPLL, the savefiles now include enough information, so parsing the log file is no longer necessary (another one of the improvements in PRPLL). The incompatibilities between PRPLL and the PrimeNet server are unrelated to the progress reporting, at least not directly. See [this post](https://www.mersenneforum.org/node/22599/page5#post899362) for the more technical details.

**Reply by preda** [reply]
There is no plan to write progress (partial progress through a test) to results.txt
There is logging being done, which includes progress, timing and error info to gpuowl-<N>.log and to standard output.
